onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.h

```diff
@@ -97,16 +99,17 @@ struct TensorrtFuncState {
   OrtMutex* tensorrt_mu_ptr = nullptr;
   bool* fp16_enable_ptr = nullptr;
   bool* int8_enable_ptr = nullptr;
+  bool* dla_enable_ptr = nullptr;
+  int* dla_core_ptr = nullptr;
   size_t* max_workspace_size_ptr = nullptr;
   std::string trt_node_name_with_precision;
-  bool engine_cache_enable;
+  bool* engine_cache_enable_ptr = nullptr;
```

Comment:
why does it need to change to pointer?
should we have done the other way around, make all the bool* to bools?  
makes the code more messy since you have to deference.
since they're primitive types, isn't it more straight forward to deal with copies rather than using references/pointers?